### DISCUSSION TITLE
Deta Base - typescript custom response

### DISCUSSION THREAD

**Discussion by mhaprean** [deta/cloud-docs#673] [Help]
Hello guys,

I want to use deta base with express js and typescript. 

```javascript

interface IUser {
  key: string;
  name: string;
  email: string;
  password: string;
  role: string;
}

const deta = Deta(process.env.DETA_PROJECT_KEY);
const usersDB = deta.Base('users');

const userResponse =  await usersDB.get('FirstUser');

```
The issue is that type of userResponse is GetResponse, but the structure is  IUser, as I inserted. I don't return the value instantly, I need to do more stuff with it, and typescript autocomplete will be very helpful. 

Currently I am doing this, but doesn't look like the right solution: 

```js
const user = (userResponse as unknown) as IUser;

```

is there a recommended way to handle this kind of things? Thank you.

**Comment by dlSean92410** [comment]
I would love to know more on this as well. 
So far this is what I'm doing
```
const res = await clientDB.get(key);
const result: Client = JSON.parse(JSON.stringify(res));
```

I think making those Base methods accept some generic type would be good. So we can do `db.get<IUser>()`.
